### PR TITLE
Fix block and page-level validation notices 

### LINKED ISSUE #1502: Should errors be shown when paired mode takes care of it?
I have noticed every post shows error in paired mode even when it's being taken care by sanitizer and frontend is perfectly valid.

Whenever you add/edit a post, there's the:
`There is content which fails AMP validation. Non-accepted validation errors prevent AMP from being served, and the user will be redirected to the non-AMP version.`

However, it's completely valid output in paired mode. It just confuses the publishers.

### PR DESCRIPTION
**Pull request by kienstra**
# Before
before-validation

# After 
accepted-notice

Fixes #1502.

### CODE REVIEW

**Comment by kienstra** [comment]
As @westonruter mentioned, the block-level validation notices didn't appear if the error was 'New Accepted'.

So this displays a block-level notice for these statuses:
* 'New Accepted
* 'New Rejected'
* 'Rejected'

...and doesn't display a notice for:
* 'Accepted'

File: assets/js/amp-block-validation.js [hunk 1]
```diff
@@ -171,8 +171,7 @@ var ampBlockValidation = ( function() { // eslint-disable-line no-unused-vars
 					return (
 						0 /* \AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_REJECTED_STATUS */ === result.status ||
 						1 /* \AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_ACCEPTED_STATUS */ === result.status ||
-						2 /* \AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_REJECTED_STATUS */ === result.status || // eslint-disable-line no-magic-numbers
-						3 /* \AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_ACCEPTED_STATUS */ === result.status // eslint-disable-line no-magic-numbers
+						2 /* \AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_REJECTED_STATUS */ === result.status // eslint-disable-line no-magic-numbers
```
Comment (westonruter) [inline assets/js/amp-block-validation.js]: This change aligns the Gutenberg editor notices with the classic editor notice:

https://github.com/Automattic/amp-wp/blob/b748977321cc3769d0476ceede56c39aa2fed267/includes/validation/class-amp-validation-manager.php#L815-L819

**Comment by kienstra** [comment]
# Paired mode with auto-sanitize on, or ‘Native’ mode
paired-auto-sanitize-or-native

# Paired mode with ‘auto_accept_sanitization’ true, but there are still non-accepted errors
auto-accept-with-rejected

# Paired or Classic mode, auto-accept disabled (unchanged in this PR)
paired-no-auto-accept

**Comment by kienstra** [comment]
**Question About Commit Above**

Hi @westonruter,
Does  https://github.com/Automattic/amp-wp/pull/1676/commits/4fea78bb3082d73cafe317153867829a1fddc230 apply your point in https://github.com/Automattic/amp-wp/issues/1502#issuecomment-428664037? 

I'm not sure if it captures what you'd like. The comment above shows the main cases it covers.

Of course, I'm also working on applying this to the block editor.

Thanks, Weston!

**Comment by kienstra** [comment]
With the latest commit (https://github.com/Automattic/amp-wp/pull/1676/commits/5afdc3eccf21769db944753ffb9fe62e5831014f):

# Native mode, only ‘Accepted’ and ‘New Accepted’ errors
native-mode-only-accepted-errors

# Native mode, with ‘Rejected’ or ‘New Rejected’ errors
native-rejected-errors

File: assets/js/amp-block-validation.js (hunk 1 above)
Comment (kienstra) [reply assets/js/amp-block-validation.js]: Nice, thanks for fixing this.

**Comment by kienstra** [comment]
# Block Editor Copy

## Native mode with only ‘Accepted’ or ‘New Accepted’ validation errors
native-mode-only-accepted

## Native Mode with a ‘Rejected’ or ‘New Rejected’ error
(Or Paired Mode, with auto-accept, with a ‘Rejected’ or ‘New Rejected’ error)
native-mode-rejected-new-rejected

## Paired mode with auto-accept disabled, with any validation error other than ‘Accepted’ (including ‘New Accepted’)
This isn't changed in this PR.
paired-mode-no-auto-accept

**Comment by kienstra** [comment]
**Request For Review**

Hi @westonruter,
Could you please review this? The 2 comments above have screenshots of the Classic and block editors. 

This might not be what you had in mind in https://github.com/Automattic/amp-wp/issues/1502#issuecomment-428664037. 

Thanks, Weston!

**Comment by westonruter** [comment]
@kienstra Overall this is looking great. I have found an issue where if there are accepted validation errors, the notice is still being displayed unexpectedly (also mentioning 0 errors):

image

**Comment by westonruter** [comment]
I can reproduce this behavior if I:

1. Add some invalid markup.
2. Save draft. 
3. Click "Review issues" link to open in new tab.
4. Accept the validation error and save the validated URL post.
5. Make another change to the post and save another draft.
6. At this point the notice is shown saying there are 0 validation errors needing to be reviewed.

**Pull request merged**